components/molecules/ContributorProfileHeader/contributor-profile-header.tsx

```diff
@@ -4,13 +4,36 @@ import RainbowBg from "img/rainbow-cover.png";
 import Button from "components/atoms/Button/button";
 import Link from "next/link";
 import { MarkGithubIcon } from "@primer/octicons-react";
+import { User } from "@supabase/supabase-js";
 
 interface ContributorProfileHeaderProps {
   avatarUrl?: string;
-  githubName?: string;
+  githubName: string;
   isConnected: boolean;
+  isFollowing: boolean;
+  handleFollow: Function;
+  handleUnfollow: Function;
+  user: User | null;
+  username: string | undefined;
 }
-const ContributorProfileHeader = ({ avatarUrl, githubName, isConnected }: ContributorProfileHeaderProps) => {
+const ContributorProfileHeader = ({
+  avatarUrl,
+  githubName,
+  isConnected,
+  isFollowing,
+  handleFollow,
+  handleUnfollow,
+  username,
+  user
+}: ContributorProfileHeaderProps) => {
+  const handleClick = () => {
+    if (isFollowing) {
+      if (handleUnfollow) handleUnfollow();
+    }
+    if (!isFollowing) {
+      if (handleFollow) handleFollow();
+    }
```

Comment:
This branching seems buggy/ prone to bugs.
an `else if(!isFollowing)` would be somewhat better. 
Here's what I think could be one of the better ways to do it:
```suggestion
    if (isFollowing) {
      if (handleUnfollow) handleUnfollow();
    }
    else if (!isFollowing) {
      if (handleFollow) handleFollow();
    }
```

Reply:
Yeah. Definitely.

> ... if we still want verbose checks:

PS: Earlier, I thought `isFollowing` had some undefined behavior (which I didn't look for) and may be that's why the verbose check's were performed.